### PR TITLE
Bugfix issue 89

### LINKED ISSUE #89: Unable to pass objects into terraform plan
Hi!
I receive the following error when passing an object into terraform plan configured with object variable:

err: could unmarshal input:
 upgrade:
  - terraform:
      vars:
        client_settings:
          create_instance: "true"
        prefix: deployment
: yaml: unmarshal errors:
  line 5: cannot unmarshal !!map into string
Error: could unmarshal input:
 upgrade:
  - terraform:
      description: Create BIG-IP Next on VIO
      vars:
        client_settings:
          create_instance: "true"
        prefix: deployment
: yaml: unmarshal errors:
  line 5: cannot unmarshal !!map into string
1 error occurred:
        * mixin execution failed: exit status 1

1 error occurred:
        * container exit code: 1, message: <nil>

Here is example porter config:

upgrade:
  - terraform:
      vars:
        prefix: "{{bundle.parameters.prefix}}"
        client_settings: {"client_settings":"true"}

Would be great if terraform mixin supported passing objects into plan.

### PR DESCRIPTION
**Pull request by sgettys**
* Allow for complex terraform input variables and output types
* Integration test covering all valid terraform input types

Fixes #89

### CODE REVIEW

**Comment by sgettys** [comment]
@carolynvs

File: build/testdata/bundles/terraform/porter.yaml [hunk 1]
```diff
@@ -4,29 +4,66 @@ mixins:
...
 registry: getporterci
+schemaVersion: 1.0.0
 
 parameters:
   - name: file_contents
     type: string
     applyTo:
       - install
       - upgrade
+  - name: map_var
```
Comment (crosbygw) [inline build/testdata/bundles/terraform/porter.yaml:15]: Might be a good idea to add defaults to verify/test yaml notation maintains quotes as expected.

File: pkg/terraform/install.go [hunk 2]
```diff
@@ -5,6 +5,7 @@ import (
 	"fmt"
 
 	"get.porter.sh/porter/pkg/exec/builder"
+	"github.com/tidwall/gjson"
```
Comment (carolynvs) [inline pkg/terraform/install.go:8]: I wanted to call out that porter already brings in a dependency on github.com/PaesslerAG/jsonpath which is pretty similar to what you are using gjson for. You may be able to reuse that dependency and not add another.

**Review by carolynvs — COMMENTED** [review]
Looks good, thanks for adding tests around other types so that we are sure the switch from -raw to -json won't be a breaking change.

File: pkg/terraform/install.go (hunk 2 above)
Comment (carolynvs) [reply pkg/terraform/install.go:8]: Turns out I was totally wrong about what y'all were doing with that library. Ignore me!

**Review by carolynvs — APPROVED** [review]
That was quick! Thanks for implementing this much requested feature. :shipit:

**Pull request merged**